src/__tests__/Search.test.js

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { fireEvent, render, screen } from "@testing-library/react";
+import Search from "../components/Search";
+
+const validProps = {
+    search: "some search text",
+    setSearchResults: jest.fn(),
+    onSubmit: jest.fn(),
+}
+
+describe("Search", () => {
+    const { asFragment } = render(<Search
+        search=""
+        setSearchResults={validProps.setSearchResults}
+        onSubmit={validProps.onSubmit} />);
+
+    it("renders correctly", () => {
+        expect(asFragment()).toMatchSnapshot()
+    });
+
+    it("renders the search button correctly", () => {
+        render(
+            <Search
+                search=""
+                setSearchResults={validProps.setSearchResults}
+                onSubmit={validProps.onSubmit}
+            />
+        );
+        const button = screen.getAllByRole("button");
+
+        expect(button).toHaveLength(1);
```

Comment:
You dont really need to do this, as youre searching for the button on line 33. Since you only have 1 button you dont need to search for a range as you are here